airflow/settings.py

```diff
@@ -107,6 +107,8 @@
     "shutdown": "blue",
 }
 
+SQL_ALCHEMY_V1 = sqlalchemy_version.startswith("1")
```

Comment:
Shoudln't that be "1." ? (It's purely hypotethical of course, but what happens when we have SqlAlchemy 10? :D

Reply:
Window ME was that one betwen 95 and 8. And well this is what happened :D

![image](https://github.com/apache/airflow/assets/595491/ddbb8ace-ea9e-4bff-b66c-9495a048034b)

Reply:
![48c](https://github.com/apache/airflow/assets/6509619/d7aa443e-1da1-4b17-9dc6-cb6665131cf8)

Sure, will fix this when rebasing.

Reply:
Since we have `packaging` available and this is a one-time calculation, I’d try to properly parse the version with `packaging.version.Version` to be the most technically correct. And maybe read the actual package version with `importlib.metadata` instead.

But even more ideally, is there a way to actually detect whether the `encoding` parameter is supported or not? Feature detection is always better than agent detection if possible (this is borrowed from web frontend but also applies well here).

Reply:
@uranusjr Thank you for the comment. This was also raised in one of the other threads: https://github.com/apache/airflow/pull/37545#discussion_r1494649517

I don't mind adding a check using `importlib` and `packaging`, but I think it shouldn't go to the sqlalchemy util, since as mentioned, we have another example of checking the dependency version similarly (pendulum). I say these checks should go into a dedicated "dependency versions" (or somesuch) utility. Thoughts?

Reply:
I don’t like putting all the checks (of different things) in one module, but maybe putting SQLA checks in the utils.sqlalchemy submodule?

Reply:
Done.